docs/guides/real-time/chat/javascript.md

```diff
@@ -75,74 +104,64 @@ document.querySelector('#new').addEventListener('submit', function (event) {
 });
 ```
 
-## Establish WebSocket Connection
-
-Within the `#login` form submit event handler, extract the `email` and `password` values from the form:
+Within the `#login` form submit event handler, get access to the email and password values
 
 ```js
 const email = event.target.elements.email.value;
 const password = event.target.elements.password.value;
 ```
 
-Create a new WebSocket, which will immediately attempt connection:
+Now, call the login method on the client, passing the email and password
 
 ```js
-connection = new WebSocket(url);
-```
+document.querySelector('#login').addEventListener('submit', function (event) {
+	event.preventDefault();
 
-On connection, you must [send an authentication message before the timeout](/guides/real-time/authentication). Add an
-event handler for the connection's `open` event:
+  const email = event.target.elements.email.value;
+  const password = event.target.elements.password.value;
 
-```js
-connection.addEventListener('open', function () {
-	connection.send(
-		JSON.stringify({
-			type: 'auth',
-			email,
-			password,
-		})
-	);
+  client.login(email, password); // [!code ++]
 });
 ```
 
-## Subscribe To Messages
-
-In a WebSocket connection, all data sent from the server will trigger the connection’s `message` event. Underneath the
-`open` event handler, add the following:
+Once the client is authenticated, immediately create a WebSocket connection:
```

Comment:
Immediately works too, because that's what's happening in this example: auth then immediately connect. I do understand your point though, in that it sounds like you must immediately connect and agree with your suggestion. 

The key difference is that this is a project-based guide so it can be more instructional.

Reply:
I get what you mean, Tim but I agree with leaving it as immediately since this is an instructional guide.